### ISSUE TITLE
Unable to send sendPasswordResetLink with TenantID

### ISSUE THREAD

**Issue by gring2** [kreait/firebase-php#573]
The auth with tenant object throw EMAIL_NOT_FOUND Error because the object do not send request with tenant id to firebase.

And I found that sendPasswordResetLink does not have tenantId case. So, i had to create workaround like below

```php
      $tenantActionSetting = new TenantIdActionSettings($tenantId);

      $auth->sendPasswordResetLink($email $tenantActionSetting);
```

But this is still very confusing that auth object with tenantID option does not support sendPassowrdEmailLink with tenantID

Do you plan to update the feature in future?

**Comment by jeromegamez** [comment]
I'm not sure this is something that can generally be handled by the SDK since we can't assume that all applications use the same variable name to store the tenant ID. The official SDKs [don't have a dedicated ActionCodeSettings field for the tenant ID either](https://firebase.google.com/docs/reference/admin/node/admin.auth.ActionCodeSettings).

Based on the information you provided I, unfortunately, can't reproduce the issue, and I don't know what you are doing with `TenantIdActionSettings`, but assuming that you add the tenant ID in the action code's URL, I think this is the way to go.

Where do you get the error? When sending the password reset link, or when clicking on the link after you received it?

I need more information 😬

**Comment by gring2** [comment]
@jeromegamez 

Thank you for your check.

I made auth object by

```php

            $filePath = 'firebase-service-account.json'
            $factory = (new Factory())->withServiceAccount($filePath);

            return $factory->withTenantId($tenantId)->createAuth();
```

and We Inject this object to everywhere we need firebase client For example signup, log in, verify-token etc.

and The problem is that We can sign up to firebase. but fail to sendResetPasswordLink.

And I think Code miss to check that auth obj have tenantID

https://cloud.google.com/identity-platform/docs/use-rest-api#section-send-password-reset-email

**Comment by jeromegamez** [comment]
Sorry for the late reply, and thank you for the detailed description 🤗! As soon as I have reproduced this with the SDK's Integration Tests, I will work on a fix, thank you for your patience!

**Comment by jeromegamez** [comment]
Thanks to your details I was able to reproduce the issue quickly and to implement a fix (181c038a0a9ed7866bc6ac4e5ed82e06d64d5615).

Some background for why I initially thought this wasn't in the scope of the SDK. In the `Auth\ApiClient`, I already included the tenant ID if available

https://github.com/kreait/firebase-php/blob/181c038a0a9ed7866bc6ac4e5ed82e06d64d5615/src/Firebase/Auth/ApiClient.php#L206-L223

But I forgot that the action link operations don't use this client (there was I time when I wanted to migrate everything in the SDK to the command pattern, but now I have both 😅)

Long story short, it should work now, and I will create a new release shortly 🤞🏼

**Comment by gring2** [comment]
@jeromegamez 
Thank you for your work. 
This SDK always helps me a lot
